climada/hazard/plot.py

```diff
@@ -65,6 +67,45 @@ def plot_rp_intensity(self, return_periods=(25, 50, 100, 250),
                                         colbar_name, title, smooth=smooth, axes=axis,
                                         figsize=figsize, adapt_fontsize=adapt_fontsize, **kwargs)
         return axis, inten_stats
+    
+    def plot_local_rp(self, threshold_intensities, smooth=True, axis=None, figsize=(9, 13), adapt_fontsize=True, cmap = 'viridis_r', **kwargs):
+        """Plot hazard local return periods for given hazard intensities.
+    
+        Parameters
+        ----------
+        threshold_intensities: np.array
+            Hazard intensities to consider for calculating return periods.
+        smooth: bool, optional
+            Smooth plot to plot.RESOLUTION x plot.RESOLUTION.
+        axis: matplotlib.axes._subplots.AxesSubplot, optional
+            Axis to use.
+        figsize: tuple, optional
+            Figure size for plt.subplots.
+        kwargs: optional
+            Arguments for pcolormesh matplotlib function used in event plots.
+    
+        Returns
+        -------
+        axis: matplotlib.axes._subplots.AxesSubplot
+            Matplotlib axis with the plot.
+        """
+        ### code to replace self._set_coords_centroids()
```

Comment:
Do we need the following six lines? (the ones replacing self._set_coords_centroids())
I mean, can it happen that the meta data exists but the centroids (or their coordinates) are not set? Also, I think one cannot manually set the centroids.geometry attribute anymore, right?

Reply:
In the latest version of Centroids (is part of develop, will be in the next release), there is no more meta attribute. Thus, I would say that you can remove these lines.

Reply:
Ok perfect.
The write_raster_... functions in io.py however use the get_meta() method. That should be fine, right?